src/core/device/swap_gpu.cc

```diff
@@ -0,0 +1,1208 @@
+    {
+        return ((struct1.ptr<struct2.ptr)||((struct1.ptr==struct2.ptr)&&(struct1.idx<struct2.idx)));
+    }
+};
+
+
+struct oneIterMsg{
+    /*
+     members: [idx, operation_type, size_delta]
+     */
+    size_t size_delta; //size if Malloc, else: delta to last index
+    int operation_type;
+    int idx;
+    oneIterMsg(size_t s, int M, int i):size_delta(s),operation_type(M),idx(i){}
+};
+
+
+struct sort_by_oneItrMsg_idx_ascending{
+    /*
+     sort oneIterMsg by Idx.
+     */
+    inline bool operator() (const oneIterMsg& struct1, const oneIterMsg& struct2)
+    {
+        return (struct1.idx<struct2.idx);
+    }
+};
+
+/// string delimiter
+vector<string> swap_split(string s, string delimiter) {
+    size_t pos_start = 0, pos_end, delim_len = delimiter.length();
+    string token;
+    vector<string> res;
+    while ((pos_end = s.find(delimiter, pos_start)) != string::npos) {
+        token = s.substr(pos_start, pos_end - pos_start);
+        pos_start = pos_end + delim_len;
+        res.push_back(token);
+    }
+    res.push_back(s.substr(pos_start));
+    return res;
+}
+
+///Section of converting text file -->vector of Sring --> pieceMsg -->pairMsg -->iterMsg
+//vector of pairMsg is used in run.
+//vector of iterMsg is used in test.
+
+vector<onePieceMsg> swap_strVec_2_pieceMsgVec(vector<string> vec, int &idxRange){
+    /*
+     convert vector of string into vector of onePieceMsg, sorted by ptr 
+     and then idx, and update idxRange to pieceMsgVec size.
+     format of onePieceMsg [ptr, size/-1, flag, idx, timestamp]
+     flag: 1 for malloc, -1 for free, 2 for read, 3 for layer,4 for mutable
+     version on 5/29, with equval blockInfo length: flag, block_, size, t
+     */
+    vector<onePieceMsg>onePieceMsgVec_;
+
+    for (int i=0;i<vec.size();i++) {
+      vector<string> v = swap_split(vec[i], " ");
+      int operation_type;
+      if (v[0]=="Malloc"){
+        operation_type = 1;
+      }else if (v[0]=="Free"){
+        operation_type = -1;
+      }else if (v[0]=="Mutable"){
+        operation_type = 4;
+      }else if (v[0]=="Read"){ 
+        operation_type = 2;
+      }else if (v[0]=="Layer"){
+        operation_type = 3;
+      }
+      //onePieceMsg(string p, size_t s, int M, int i):ptr(p),size(s),operation_type(M),idx(i){}
+      size_t result;
+      stringstream convert(v[2]);
+      if (!(convert>>result)){
+          result =-1;
+          cout<<"error for converting size from str to int."<<endl;
+      }
+      onePieceMsg tempMsg(v[1],result, operation_type, i);
+      double temp_time;
+      stringstream convert2(v[3]);
+      convert2>>temp_time;
+      tempMsg.t =temp_time;
+      onePieceMsgVec_.push_back(tempMsg);
+    }
+ 
+    sort(onePieceMsgVec_.begin(),onePieceMsgVec_.end(),sort_by_ptr_idx_ascending());
+    idxRange = static_cast<int>(onePieceMsgVec_.size());
+
+    return onePieceMsgVec_;
+}// end of strVec_2_pieceMsgVec function
```

Comment:
this comment is meaningless..
better add some comments for the functionality or the arguments.